lib/googleauth.rb

```diff
@@ -52,15 +52,36 @@ class DefaultCredentials
 
       # override CredentialsLoader#make_creds to use the class determined by
       # loading the json.
-      def self.make_creds(json_key_io, scope = nil)
-        json_key, clz = determine_creds_class(json_key_io)
-        clz.new(StringIO.new(MultiJson.dump(json_key)), scope)
+      def self.make_creds(json_key_io: nil, scope: nil)
+        if json_key_io
+          json_key, clz = determine_creds_class(json_key_io)
+          clz.new(json_key_io: StringIO.new(MultiJson.dump(json_key)),
+                  scope: scope)
+        else
+          clz = read_creds
+          clz.new(scope: scope)
+        end
+      end
+
+      def self.read_creds
+        env_var = CredentialsLoader::ACCOUNT_TYPE_VAR
+        fail "#{ACCOUNT_TYPE_VAR} is undefined in env" if ENV[env_var].nil?
+        type = ENV[env_var]
+        case type
+        when 'service_account'
+          ServiceAccountCredentials
+        when 'authorized_user'
+          UserRefreshCredentials
+        else
+          fail "credentials type '#{type}' is not supported"
+        end
       end
 
       # Reads the input json and determines which creds class to use.
       def self.determine_creds_class(json_key_io)
         json_key = MultiJson.load(json_key_io.read)
-        fail "the json is missing the #{key} field" unless json_key.key?('type')
+        key = 'type'
+        fail "the json is missing the #{key} field" unless json_key.key?(key)
         type = json_key['type']
```

Comment:
`type = json_key['type']` -> `type = json_key[key]` and `"the json is missing the #{key} field"` to `"the json is missing the '#{key}' field"`